### PR TITLE
squall/clausematch: add flag to indicate interest in archived object

### PR DESCRIPTION
**Pull request by malkhamis-panw**
This PR is needed so the user can indicate if they are interested in archived objects for endpoint `clausesmatch`. For changes to make more sense, please refer to this [project](https://github.com/orgs/aporeto-inc/projects/1173).

### CODE REVIEW

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "clausematch-includes-archived-objects",
    "component": "gaia",
    "pr-id": "733",
    "commit-sha": "38659a7eb33e5cd435884bae7638ff0377026e3d",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "underwater",
    "pr-id": "178",
    "commit-sha": "2c730a32ad0b1b29a63fdfb82dc682f33c0e8c0d",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "backend",
    "pr-id": "1433",
    "commit-sha": "20cfd800e24faaba815000a3c0523563c9234560",
    "pipeline": "master"
  }
]
```

File: specs/clausesmatch.spec [hunk 1]
```diff
@@ -26,6 +26,14 @@ attributes:
     validations:
     - $tagsExpression
 
+  - name: includeArchived
```
Comment (t00f) [inline specs/clausesmatch.spec:29]: usually we call it just `archived`. Let's keep it consistent with the rest of the API

Comment (malkhamis-panw) [reply specs/clausesmatch.spec:29]: I thought that would be confusing as "only archived".. but if that is the convention, I can follow it

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "clausematch-includes-archived-objects",
    "component": "gaia",
    "pr-id": "733",
    "commit-sha": "37af9a6ba5c16fab2c045efb4623365791622aa0",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "underwater",
    "pr-id": "178",
    "commit-sha": "2c730a32ad0b1b29a63fdfb82dc682f33c0e8c0d",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "backend",
    "pr-id": "1433",
    "commit-sha": "9960ea9c703f30caf52e34cf475b1ecd3a3dea87",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "clausematch-includes-archived-objects",
    "component": "underwater",
    "pr-id": "178",
    "commit-sha": "2c730a32ad0b1b29a63fdfb82dc682f33c0e8c0d",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "backend",
    "pr-id": "1433",
    "commit-sha": "6d18b6dea3df3b8156e29af65873bb81c3893318",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "gaia",
    "pr-id": "733",
    "commit-sha": "37af9a6ba5c16fab2c045efb4623365791622aa0",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "clausematch-includes-archived-objects",
    "component": "underwater",
    "pr-id": "178",
    "commit-sha": "2c730a32ad0b1b29a63fdfb82dc682f33c0e8c0d",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "backend",
    "pr-id": "1433",
    "commit-sha": "6d18b6dea3df3b8156e29af65873bb81c3893318",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "gaia",
    "pr-id": "733",
    "commit-sha": "c6ecc92681170e95ff9af39462d570388f57b48d",
    "pipeline": "master"
  }
]
```

**Comment by malkhamis-panw** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "clausematch-includes-archived-objects",
    "component": "underwater",
    "pr-id": "178",
    "commit-sha": "fef8fe6b14628fe9253644f15b0d6fd85c06fc1b",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "backend",
    "pr-id": "1433",
    "commit-sha": "6d18b6dea3df3b8156e29af65873bb81c3893318",
    "pipeline": "master"
  },
  {
    "project": "clausematch-includes-archived-objects",
    "component": "gaia",
    "pr-id": "733",
    "commit-sha": "c6ecc92681170e95ff9af39462d570388f57b48d",
    "pipeline": "master"
  }
]
```

**Pull request merged**